clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -162,6 +164,11 @@ private void stop(String errorMessage) {
         transport.stop();
         connectionState = HubConnectionState.DISCONNECTED;
         logger.log(LogLevel.Information, "HubConnection stopped.");
+        if (onClosedCallbackList != null){
+            for (Consumer<Exception> callback: onClosedCallbackList) {
+                callback.accept(new HubException(errorMessage));
```

Comment:
Move `new HubException(errorMessage)` above for loop

Reply:
What happens if you call start in the close callback?

Reply:
It will try to restart the connection